drivers/gpio/gpio_altera_pio.c

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright (c) 2022, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT altr_pio_1_0
+
+/**
+ * @file
+ * @brief PIO driver for Intel FPGA PIO Core IP
+ * Reference : Embedded Peripherals IP User Guide : 27. PIO Core
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#define ALTERA_AVALON_PIO_DATA_OFFSET		0x00
+#define ALTERA_AVALON_PIO_DIRECTION_OFFSET	0x04
+#define ALTERA_AVALON_PIO_IRQ_OFFSET		0x08
+#define ALTERA_AVALON_PIO_SET_BITS		0x10
+#define ALTERA_AVALON_PIO_CLEAR_BITS		0x14
+
+typedef void (*altera_cfg_func_t)(void);
+
+struct gpio_altera_config {
+	struct gpio_driver_config	common;
+	uintptr_t			reg_base;
+	uint32_t			irq_num;
+	uint8_t				direction;
+	altera_cfg_func_t		cfg_func;
+};
+
+struct gpio_altera_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+	/* list of callbacks */
+	sys_slist_t cb;
+};
+
+/**
+ * @brief Determine the pin direction.
+ *
+ * @param dev		Pointer to device structure for the driver instance.
+ * @param pin_mask	Value indicating which pins will be determined.
+ *
+ * @return 0 if the pin direction is input, 1 if the pin direction is output.
+ */
```

Comment:
Please remove doxygen that duplicates (and possibly conflicts with) public driver api doxygen.